src/Compilers/CSharp/Test/Emit/Emit/EmitMetadataTests.cs

```diff
@@ -2936,5 +2936,329 @@ public class Child : Parent, IParent
+                	{
+                		// Fields
+                		.field assembly static initonly string s
+                		// Methods
+                		.method private hidebysig specialname rtspecialname static 
+                			void .cctor () cil managed 
+                		{
+                			// Method begins at RVA 0x2089
+                			// Code size 17 (0x11)
+                			.maxstack 8
+                			IL_0000: ldsflda valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '<PrivateImplementationDetails>'::'185F8DB32271FE25F561A6FC938B2E264306EC304EDA518007D1764826381969'
+                			IL_0005: ldc.i4.5
+                			IL_0006: call string '<PrivateImplementationDetails>'::BytesToString(uint8*, int32)
+                			IL_000b: stsfld string '<PrivateImplementationDetails>'/'<S>1BFD09D1A433FB78117B4C7B1583D16D'::s
+                			IL_0010: ret
+                		} // end of method '<S>1BFD09D1A433FB78117B4C7B1583D16D'::.cctor
+                	} // end of class <S>1BFD09D1A433FB78117B4C7B1583D16D
+                	// Fields
+                	.field assembly static initonly valuetype '<PrivateImplementationDetails>'/'__StaticArrayInitTypeSize=5' '185F8DB32271FE25F561A6FC938B2E264306EC304EDA518007D1764826381969' at I_{{offset}}
+                    .data cil I_{{offset}} = bytearray (
+                		48 65 6c 6c 6f
+                	)
+                	// Methods
+                	.method private hidebysig static 
+                		string BytesToString (
+                			uint8* bytes,
+                			int32 length
+                		) cil managed 
+                	{
+                		// Method begins at RVA 0x207b
+                		// Code size 13 (0xd)
+                		.maxstack 8
+                		IL_0000: call class [mscorlib]System.Text.Encoding [mscorlib]System.Text.Encoding::get_UTF8()
+                		IL_0005: ldarg.0
+                		IL_0006: ldarg.1
+                		IL_0007: callvirt instance string [mscorlib]System.Text.Encoding::GetString(uint8*, int32)
+                		IL_000c: ret
+                	} // end of method '<PrivateImplementationDetails>'::BytesToString
+                } // end of class <PrivateImplementationDetails>
+                """);
+        }
+
+        [Theory, CombinatorialData]
+        public void DataSectionStringLiterals_MetadataOnly(
+            [CombinatorialValues("0", "off")] string feature)
+        {
+            var source = """
+                System.Console.WriteLine("Hello");
+                """;
+            CompileAndVerify(source,
+                emitOptions: EmitOptions.Default.WithEmitMetadataOnly(true),
+                parseOptions: TestOptions.Regular.WithFeature("experimental-data-section-string-literals", feature),
+                verify: Verification.FailsPEVerify with
+                {
+                    PEVerifyMessage = """
+                        Bad token as entry point in CLR header.
```

Comment:
>Bad token as entry point in CLR header. [](http://example.com/codeflow?start=24&length=39)

What does this complain about?

Reply:
I'm not sure how to get more details about the error, but it's caused by the metadata-only emit (`EmitOptions.Default.WithEmitMetadataOnly(true)`) so I suspect some non-metadata parts of the DLL are missing and PEVerify doesn't like that.

Reply:
Is this error reported in scenarios that do not utilize the feature? For example, are there existing tests observing the same failure? Would it be reported when explicit `Main` is used and we emit EXE or DLL? It is quite possible that the feature has nothing to do with the error, but the error might still be something to be concerned about.

Reply:
> Is this error reported in scenarios that do not utilize the feature?

Yes, this test verifies the scenario with both the feature on and off.

> Would it be reported when an explicit `Main` is used and we emit EXE or DLL?

That seems to be the reason, I will update the test. Also filed https://github.com/dotnet/roslyn/issues/76707.